front/app/containers/Admin/projects/project/nativeSurvey/FormResults/FormResultsQuestion/components/ViewSingleSubmissionNotice.tsx

```diff
@@ -34,42 +32,63 @@ const ViewSingleSubmissionNotice = ({ customFieldId }: Props) => {
     projectId: phaseId ? undefined : projectId,
     phaseId,
   });
-
-  const additionalCustomFieldsIds = useRelevantToHeatmapInputCustomFields({
+  const { data: inputCustomFields } = useCustomFields({
     projectId,
     phaseId,
-  })?.map((customField) => customField.id);
+  });
+
+  const inputCustomFieldsIds = inputCustomFields?.map(
+    (customField) => customField.id
+  );
 
   const relevantAnalysis = analyses?.data.find(
-    (analysis) =>
-      analysis.relationships.main_custom_field?.data?.id === customFieldId
+    (analysis) => analysis.relationships.main_custom_field?.data === null
   );
 
+  const analysisCustomFieldIds =
+    relevantAnalysis?.relationships.additional_custom_fields?.data.map(
+      (field) => field.id
+    ) || [];
+
+  const customFieldsMatchAnalysisAdditionalFields =
+    inputCustomFieldsIds &&
+    analysisCustomFieldIds.length === inputCustomFieldsIds.length &&
+    analysisCustomFieldIds.every((id) => inputCustomFieldsIds.includes(id));
+
+  const openAnalysis = (analysisId: string) => {
+    clHistory.push(
+      `/admin/projects/${projectId}/analysis/${analysisId}?${stringify({
+        phase_id: phaseId,
+      })}`
+    );
+  };
+
   const goToAnalysis = () => {
     if (relevantAnalysis?.id) {
-      clHistory.push(
-        `/admin/projects/${projectId}/analysis/${
-          relevantAnalysis.id
-        }?${stringify({
-          phase_id: phaseId,
-        })}`
-      );
+      if (!customFieldsMatchAnalysisAdditionalFields) {
+        updateAnalysis(
```

Comment:
If the custom fields in the current analysis differ from the custom fields in the form (they've added a field, for example), the analysis gets automatically updated to include the new question. This creates a more seemless interface where we don't put it on the users to manually add the new questions and I think it works well.